src/rgw/rgw_sal_sfs.h

```diff
@@ -327,25 +327,59 @@ class SFStore : public Store {
   }
 
   sfs::BucketRef bucket_create(
-    const rgw_bucket &bucket, const RGWUserInfo &owner
+    const rgw_bucket &bucket,
+    const RGWUserInfo &owner,
+    const std::string &zonegroup_id,
+    const rgw_placement_rule &placement_rule,
+    const std::string &swift_ver_location,
+    const RGWQuotaInfo *pquota_info,
+    std::map<std::string, bufferlist> &attrs,
+    RGWBucketInfo &info,
+    obj_version &ep_objv
   ) {
     std::lock_guard l(buckets_map_lock);
     if (_bucket_exists(bucket.name)) {
       return nullptr;
     }
 
-    sfs::sqlite::DBOPBucketInfo info;
-    info.binfo.bucket = bucket;
-    info.binfo.owner = owner.user_id;
-    info.binfo.creation_time = ceph::real_clock::now();
-    info.binfo.placement_rule.name = "default";
-    info.binfo.placement_rule.storage_class = "STANDARD";
+    sfs::sqlite::DBOPBucketInfo db_binfo;
+    db_binfo.binfo.bucket = bucket;
+    db_binfo.binfo.owner = owner.user_id;
+    db_binfo.binfo.creation_time = ceph::real_clock::now();
+    db_binfo.binfo.placement_rule = placement_rule;
+    db_binfo.binfo.zonegroup = zonegroup_id;
+
+    RGWObjVersionTracker& objv_tracker = info.objv_tracker;
+    objv_tracker.read_version.clear();
+    objv_tracker.generate_new_write_ver(cctx);
+    objv_tracker.read_version = objv_tracker.write_version;
```

Comment:
`RGWObjVersionTracker` is not stored in `sqlite` yet, same as `zonegroup`.

I'm happy to merge this without storing yet as this is part of a bug fix for the UI but when we merge this branch we should open a bug so we keep track and remember to store them.
For future API calls, when the bucket is already in sqlite and not in memory those fields will be missing.

Reply:
Yes, you are right, I will open an issue for this.
I have been reluctant to modify the sqlite table on this iteration because probably you are already modifying it and I wanted to avoid conflicts.